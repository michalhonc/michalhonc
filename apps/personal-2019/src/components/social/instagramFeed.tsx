import * as React from "react";
import axios from "axios";

import { SocialWrapper } from "./socialWrapper";
import { State, useFetch } from "../../hooks/useFetch";
import Instagram from "../../images/instagram.svg";
import Like from "../../images/like.svg";
import Comment from "../../images/comment.svg";

interface IInstagram {
  big: string;
  small: string;
  url: string;
  caption: string;
  likes: number;
  comments: number;
  id: number;
}

export const InstagramFeed = () => {
  const { posts, state } = useFetch<IInstagram>("getInstagramPosts");

  if (state === State.Err) return null;

  return (
    <SocialWrapper
      Icon={Instagram}
      profile="@michalhonc.cz"
      profileUrl="https://www.instagram.com/michalhonc.cz/"
      className="InstagramFeed"
    >
      {state === State.Loading &&
        [...Array(6)].map((_, i) => (
          <a href="#" key={i}>
            <div className="InstagramFeed-imageWrapper--fake" />
          </a>
        ))}

      {posts.map((post: IInstagram) => (
        <a href={post.url} key={post.id}>
          <div className="InstagramFeed-imageWrapper">
            <img
              loading="lazy"
              width="200"
              height="200"
              src={post.small}
              alt="@michalhonc.cz Instagram profile"
            />
          </div>
          <div className="InstagramFeed-meta">
            {post.likes > 0 && (
              <span>
                {post.likes} <Like />
              </span>
            )}
            {post.comments > 0 && (
              <span>
                {post.comments} <Comment />
              </span>
            )}
          </div>
        </a>
      ))}
    </SocialWrapper>
  );
};
