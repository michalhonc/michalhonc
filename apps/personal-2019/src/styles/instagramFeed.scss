@keyframes loading {
  to {
    background-position: 350% 0;
  }
}

.InstagramFeed {
  display: flex;
  flex-wrap: wrap;
  margin: -0.5rem;

  @include respond(phone) {
    justify-content: center;
  }

  & a {
    position: relative;
    margin: 0.5rem;
  }

  &-imageWrapper {
     display: flex;
     align-items: flex-start; // iOS sets align-items: streched which causes bug
     overflow: hidden;

    &--fake {
      width: 20rem;
      height: 20rem;
      background-color: var(--c-alpha);
    }

  }

  & img {
    min-height: 100%;
    width: auto;
    transition: all 0.2s;

    &:hover {
      transform: scale(1.1) rotate(1deg);
    }

    @include respond(phone) {
      height: auto;
      max-width: 15rem;
    }
  }

  &-meta {
    position: absolute;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    bottom: 0;
    left: 0;
    height: 3rem;
    background: rgba(4, 119, 106, 0.4);
    padding: 0 2rem 0 1rem;
    border-radius: 0 2rem 0 0;

    & span {
      display: flex;
      font-size: 1.2rem;
      line-height: 1.6rem;
      font-weight: bold;
      color: var(--c-white);

      &:not(:last-of-type) {
        margin-right: 1.5rem;
      }
    }

    & svg {
      width: 1.6rem;
      height: 1.6rem;
      fill: var(--c-white);
      margin-left: 0.5rem;
    }
  }
}
