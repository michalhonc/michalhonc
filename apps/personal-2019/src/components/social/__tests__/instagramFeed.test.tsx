import * as React from "react";
import renderer from "react-test-renderer";
import { InstagramFeed } from "../instagramFeed";

describe("instagramFeed", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<InstagramFeed />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
