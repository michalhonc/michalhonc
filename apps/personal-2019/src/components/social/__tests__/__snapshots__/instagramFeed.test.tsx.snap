// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instagramFeed renders correctly 1`] = `
<div
  className="Social-wrapper"
>
  <div
    className="Social-icon"
  >
    <test-file-stub />
  </div>
  <a
    className="underline Social-profile"
    href="https://www.instagram.com/michalhonc.cz/"
  >
    @michalhonc.cz
  </a>
  <div
    className="Social-content InstagramFeed"
  />
</div>
`;
